registry/interface.go

```diff
@@ -9,6 +9,9 @@ import (
 
 // SignatureRepository provides a storage for signatures
 type SignatureRepository interface {
+	// GetCoreSignatureManifest returns signature manifests for verification filterd by given subject digest
+	GetCoreSignatureManifest(ctx context.Context, subjectDigest digest.Digest) ([]CoreSignatureManifest, error)
+
 	// Lookup finds all signatures for the specified manifest
 	Lookup(ctx context.Context, manifestDigest digest.Digest) ([]digest.Digest, error)
```

Comment:
How about `List` to replace `GetCoreSignatureManifest` and `Lookup`, listing all signatures for the specified manifest? Will that sound better?

```go
import artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
```

```go
List(ctx context.Context, manifestDigest digest.Digest) ([]SignatureManifest, error)
```

Reply:
The renaming makes sense to me. My only concern is that the returned `SignatureManifest` which only includes `blob` and `annotations` fileds. It's like a partial `SignatureManifest`. What if we need an api that returns a complete `SignatureManifest` in the future? Do we want to fill all fields in this struct right now or give it a different name?

Reply:
Synced offline. Created `SignatureManifest` struct including a `blob` and `annotations`. And refactored `List` to return a slice of `SignatureManifest`.